src/eureka/S1_detector_processing/bias_sub.py

```diff
@@ -0,0 +1,187 @@
+
+    # Replace NaN's in the superbias with zeros
+    bias_model.data[np.isnan(bias_model.data)] = 0.0
+
+    # Create 4D bias array for each integration/group and apply scale factor 
+    bias_data = np.ones((nint, ngroup, nrow, ncol)) * \
+        bias_model.data[np.newaxis, np.newaxis, :, :]
+    
+    try:
+        if meta.bias_correction == "mean":
+            if isinstance(meta.bias_group, int):
+                # Scale superbias frame using single value for all groups
+                jj = meta.bias_group
+                assert jj > 0, f"Group number should be >0, got: {jj}"
+                assert jj <= ngroup, f"Group number should be <={ngroup}" + \
+                    ", got: {jj}"
+                log.writelog('  Applying mean bias correction using ' +
+                             f'group {jj}.')
+                bias_model.data *= mean_scale_factor[jj-1]
+            elif meta.bias_group == "each":
+                # Scale superbias frame using means values for each group
+                log.writelog('  Applying mean bias correction to each group.')
+                bias_model.data *= mean_scale_factor[0]
+                bias_data *= mean_scale_factor[np.newaxis, :, np.newaxis, 
+                                               np.newaxis]
+            else:
+                raise ValueError('Incorrect meta.bias_group value: ' + 
+                                 f'{meta.bias_group}. Should be ' +
+                                 '[1, 2, ..., each].')
+        elif (meta.bias_correction == "group_level") or \
+             (meta.bias_correction == "smooth"):
+            # Apply correction for zeroframe
+            bias_model.data *= mean_scale_factor[0]
+            if isinstance(meta.bias_group, int):
+                # Scale superbias frame using values from one group
+                jj = meta.bias_group
+                assert jj > 0, f"Group number should be >0, got: {jj}"
+                assert jj <= ngroup, f"Group number should be <={ngroup}" + \
+                    ", got: {jj}"
+                log.writelog('  Applying group-level bias correction using ' +
+                             f'group {jj}.')
+                bias_data *= scale_factor[:, jj-1, np.newaxis, 
+                                          np.newaxis, np.newaxis]
+            elif meta.bias_group == "each":
+                # Scale superbias frame using values from each group
+                log.writelog('  Applying group-level bias correction to ' + 
+                             'each group.')
+                bias_data *= scale_factor[:, :, np.newaxis, np.newaxis]
+            else:
+                raise ValueError('Incorrect meta.bias_group value: ' + 
+                                 f'{meta.bias_group}. Should be ' +
+                                 '[1, 2, ..., each].')
+        else:
+            log.writelog('  No bias correction applied.')
+            bias_data = None
+    except Exception as error:
```

Comment:
Do we really want to catch these exceptions? Does the code just fail more gracefully if the exceptions are caught and None is returned, or are you wanting the code to continue on without the user's requested effects? Because there is so much text logged to the console during Stage 1, I can easily imagine someone missing the error messages printed to the console and assuming everything ran successfully if this causes the code not to crash when an error is hit

Reply:
I'm trying to catch an assortment of possible errors within the `try` statement and provide meaningful error messages.  When returning `None`, the code crashes on setting the step to complete, which works out well.

Reply:
1. So this screenshot is what I get if I run your new code with `bias_group=-1` to try to throw an error:
<img width="1612" alt="Screen Shot 2023-03-07 at 14 50 32" src="https://user-images.githubusercontent.com/24514122/223576759-8adc7ce8-0c3d-48f5-a13a-f7ebb83742ec.png">

2. If we keep the try+except for the purposes of putting the error into the log, but then do `raise error` instead of `return None`:
<img width="1612" alt="Screen Shot 2023-03-07 at 15 13 44" src="https://user-images.githubusercontent.com/24514122/223577312-592b12ee-a639-4255-9c88-80232ae66a73.png">

3. Meanwhile, if we remove the try+except, then we get:
<img width="1612" alt="Screen Shot 2023-03-07 at 15 18 20" src="https://user-images.githubusercontent.com/24514122/223577140-8806659b-ada9-4822-9006-8b2b88d6d250.png">


People so rarely read further up into their tracebacks, so whatever error ends up on that bottom line is the one they're most likely to see. I think we should go with option 2 or 3, dependeing on whether you want the error in the log file.

Reply:
Ahh, yes.  I'll use `raise error`